components/kyma-environment-broker/internal/orchestration/manager/upgrade_kyma.go

```diff
@@ -60,6 +61,9 @@ func (u *upgradeKymaFactory) NewOperation(o internal.Orchestration, r orchestrat
 		},
 		SMClientFactory: u.smcf,
 	}
+	if o.Parameters.Kyma.Version != "" {
```

Comment:
why do we need this? let's discuss

Reply:
maybe we have to compute the version according to the GA/SA->version mapping and the default one.

Reply:
ok, I see the reason. It should be nice to cover the scenario with manual version set in the orchestration by the orchestration test. let's discuss it

Reply:
OK, will create a test case for that scenario

Reply:
The main reason for this change is that I wanted to eliminate orchestration usage in upgrade operation's initialization step. It is a bad pattern introduced lately in by my colleague, and wanted to eliminate that. Operation data MUST enclose all data needed to perform the operation, it should not "fetch" some shadow data from parent orchestration.

Reply:
Scenerio covered in a new orchestration test case in latest commit